### PR TITLE
POLICY-139 Added new API for moving assets from governed to non-compliant and vice-versa

### PR DESCRIPTION
**Pull request by arpit-at**
## Change description

> Added new API for moving assets from governed to non-compliant and vice-versa required for policy manager

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues
- https://atlanhq.atlassian.net/browse/POLICY-139

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: webapp/src/main/java/org/apache/atlas/web/rest/BusinessPolicyREST.java [hunk 1]
```diff
@@ -112,4 +113,45 @@ public void unlinkBusinessPolicy(@PathParam("policyId") final String policyGuid,
...
+
+    @POST
+    @Path("/move/{assetId}")
+    @Timed
+    public void moveBusinessPolicy(
+            @PathParam("assetId") String assetId,
+            final MoveBusinessPolicyRequest request
+    ) throws AtlasBaseException {
+        // Start performance metric recording
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("moveBusinessPolicy");
```
Comment (PRATHAM2002-DS) [inline webapp/src/main/java/org/apache/atlas/web/rest/BusinessPolicyREST.java:126]: We can use only perf logs here which is already enabled in lines below, and remove the metrics logs.

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java [hunk 2]
```diff
@@ -5044,6 +5036,52 @@ private void updateDomainAttribute(AtlasVertex vertex, Set<String> existingValue
         existingValues.forEach(existingValue -> vertex.removePropertyValue(DOMAIN_GUIDS_ATTR, existingValue));
         vertex.setProperty(DOMAIN_GUIDS_ATTR, meshEntityId);
     }
+    public AtlasVertex moveBusinessPolicy(Set<String> policyIds, String assetId, String type) {
+        // Retrieve the AtlasVertex for the given assetId
+        AtlasVertex assetVertex = AtlasGraphUtilsV2.findByGuid(graph, assetId);
```
Comment (PRATHAM2002-DS) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:5041]: I think add a null check on assetVertex.

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java [hunk 3]
```diff
@@ -5044,6 +5036,52 @@ private void updateDomainAttribute(AtlasVertex vertex, Set<String> existingValue
...
+        // Update the modification metadata
+        updateModificationMetadata(assetVertex);
+
+        // Create a differential AtlasEntity to reflect the changes
+        AtlasEntity diffEntity = new AtlasEntity(assetVertex.getProperty(TYPE_NAME_PROPERTY_KEY, String.class));
+        diffEntity.setGuid(assetVertex.getProperty(GUID_PROPERTY_KEY, String.class));
+        diffEntity.setAttribute(ASSET_POLICY_GUIDS, governedPolicies);
+        diffEntity.setAttribute(NON_COMPLIANT_ASSET_POLICY_GUIDS, nonCompliantPolicies);
+        diffEntity.setUpdatedBy(assetVertex.getProperty(MODIFIED_BY_KEY, String.class));
+        diffEntity.setUpdateTime(new Date(RequestContext.get().getRequestTime()));
```
Comment (PRATHAM2002-DS) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java:5078]: You can use this function `setEntityCommonAttributes` to set common attributes.

File: webapp/src/main/java/org/apache/atlas/web/rest/BusinessPolicyREST.java (hunk 1 above)
Comment (arpit-at) [reply webapp/src/main/java/org/apache/atlas/web/rest/BusinessPolicyREST.java:126]: so to trace individual method metrics we don't need metrics? Can perf log give a metric for all the methods used called from this as well?

**Review by PRATHAM2002-DS — APPROVED** [review]

**Pull request merged**